base/test/testSignalGeneratorSrc_tests.cpp

```diff
@@ -2,76 +2,88 @@
 #include "Module.h"
 #include "RawImageMetadata.h"
 #include <boost/test/unit_test.hpp>
-#include "FileWriterModule.h"
 #include <boost/filesystem.hpp>
 #include "PipeLine.h"
-#include "StatSink.h"
 #include "test_utils.h"
 #include "ExternalSinkModule.h"
+#include "FrameContainerQueue.h"
 
 BOOST_AUTO_TEST_SUITE(TestSignalGenerator_tests)
 
+class SinkModuleProps : public ModuleProps
+{
+public:
+    SinkModuleProps() : ModuleProps(){};
+};
+
+class SinkModule : public Module
+{
+public:
+    SinkModule(SinkModuleProps props) : Module(SINK, "sinkModule", props){};
+    boost::shared_ptr<FrameContainerQueue> getQue() { return Module::getQue(); }
+
+protected:
+    bool validateOutputPins()
+    {
+        return true;
+    }
+    bool validateInputPins()
+    {
+        return true;
+    }
+};
+
 BOOST_AUTO_TEST_CASE(Basic)
 {
-    auto source = boost::shared_ptr<TestSignalGenerator>(new TestSignalGenerator(TestSignalGeneratorProps(640, 360)));
-    auto metadata = framemetadata_sp(new RawImagePlanarMetadata(640, 360, ImageMetadata::ImageType::YUV420, size_t(0), CV_8U));
+    auto source = boost::shared_ptr<TestSignalGenerator>(new TestSignalGenerator(TestSignalGeneratorProps(400, 400)));
+    auto metadata = framemetadata_sp(new RawImagePlanarMetadata(400, 400, ImageMetadata::ImageType::YUV420, size_t(0), CV_8U));
     source->addOutputPin(metadata);
```

Comment:
addOutputPin is done externally in pipeline in the cases where the module does not know the output pin by itself, for eg. FileReader. That is not the case here. Here, if resolution is given as prop, the output pin is going to always be YUV420 , 8 bit image with same resolution. So, the addOutputPin() should be done inside the module. Not to be done by user in the pipeline.